google/cloud/pubsub/samples/pubsub_samples_common.cc

```diff
@@ -159,6 +160,19 @@ std::string RandomSchemaId(google::cloud::internal::DefaultPRNG& generator) {
                                                        "cloud-cpp-samples");
 }
 
+StatusOr<std::string> ReadFile(std::string const& path) {
+  std::ifstream ifs(path);
+  if (!ifs.is_open()) {
+    return google::cloud::Status{google::cloud::StatusCode::kNotFound,
+                                 "can't find file: " + path};
+  }
+
+  std::stringstream buffer;
+  buffer << ifs.rdbuf();
+  ifs.close();
+  return buffer.str();
+}
+
```

Comment:
We use exceptions in samples, you can "simplify" this code to something like:

```suggestion
std::string ReadFile(std::string const& path) {
  std::ifstream ifs(path);
  if (!ifs.is_open()) throw std::runtime_error("Cannot open file: " + path);
  ifs.exceptions(std::ios::badbit);
  return std::string{std::istreambuf_iterator<char>{ifs.rdbuf()}, {}};
}
```

That will make the samples themselves easier to grok, because they will need less error-handling code.